### PR TITLE
Set back alias related elements and update to marlon version

### PR DESCRIPTION
**Pull request by b-alidra**
Set back the deleted alias related elements and, by the way, updated to [marlon-be](https://github.com/marlon-be/marlon-ogone/commits/master) version to get the last improvements.

### CODE REVIEW

**Comment by wysow** [comment]
Thx a lot @b-alidra for this, is it possible to squash all this in one commit?

I will activate testing and other stuff right after so that we can be confident to realease version 1 of this lib.

File: README.md [hunk 1]
```diff
@@ -1,33 +1,47 @@
-# Ogone PHP library #
+# PostFinance PHP library #
 
-This library allows you to easily implement an [Ogone](http://ogone.com) integration into your project.
-It provides the necessary components to complete a correct payment flow with the [Ogone](http://ogone.com) platform.
-
-[![Scrutinizer Quality Score](https://scrutinizer-ci.com/g/marlon-be/marlon-ogone/badges/quality-score.png?s=ceabe3e767cad807b589fc63169e6a330d20f9fa)](https://scrutinizer-ci.com/g/marlon-be/marlon-ogone/)
-[![Build Status](https://travis-ci.org/marlon-be/marlon-ogone.png)](https://travis-ci.org/marlon-be/marlon-ogone)
+This library allows you to easily implement an [PostFinance](https://www.postfinance.ch/) integration into your project.
+It provides the necessary components to complete a correct payment flow with the [PostFinance](https://www.postfinance.ch/i) platform.
```
Comment (wysow) [inline README.md]: `https://www.postfinance.ch/` instead of `https://www.postfinance.ch/i`

File: README.md [hunk 2]
```diff
@@ -1,33 +1,47 @@
...
 As always, this is work in progress. Please feel free to fork this project and let them pull requests coming!
 
+Installation:
+-------------
+The library is [PSR-0 compliant](http://www.php-fig.org/psr/psr-0/fr/)
+ and the simplest way to install it is via composer, simply add:
+
+    {
+        "require": {
+            "ba/php-ticons": "dev-master"
```
Comment (wysow) [inline README.md]: Wrong reference here...

And should just be: `composer require wysow/postfinance`, no need to manually edit `composer.json` file and run `composer install`.

File: README.md [hunk 3]
```diff
@@ -1,33 +1,47 @@
...
+        }
+    }
+
+into your composer.json, then run 'composer install' or 'composer update' as required.
+
 ## Overview ##
 
-The library complies to the [PSR-0 standard](http://groups.google.com/group/php-standards/web/psr-0-final-proposal),
+The library complies to the [PSR-0 standard](http://www.php-fig.org/psr/psr-0/fr/),
 so it can be autoloaded using PSR-0 classloaders like the one in Symfony2. See autoload.php for an example.
```
Comment (wysow) [inline README.md]: `autoload.php` file as been deleted so this should be also deleted.

File: composer.json [hunk 4]
```diff
@@ -1,10 +1,14 @@
 {
-    "name": "wysow/postfinance",
+    "name": "balidra/postfinance",
```
Comment (wysow) [inline composer.json]: This is not possible, you cannot change the package name as Packagist already knows it as `wysow/postfinance` and it should not be duplicated...

File: lib/PostFinance/DirectLink/DirectLinkMaintenanceRequest.php [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+namespace PostFinance\DirectLink;
+
+use PostFinance\AbstractDirectLinkRequest;
+use PostFinance\ShaComposer\ShaComposer;
+use InvalidArgumentException;
+
+class DirectLinkMaintenanceRequest extends AbstractDirectLinkRequest {
+
+    const TEST = "https://secure.ogone.com/ncol/test/maintenancedirect.asp";
+    const PRODUCTION = "https://secure.ogone.com/ncol/prod/maintenancedirect.asp";
```
Comment (wysow) [inline lib/PostFinance/DirectLink/DirectLinkMaintenanceRequest.php]: This 2 urls should be updated to the Postinance ones...

File: lib/PostFinance/DirectLink/DirectLinkPaymentRequest.php [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+namespace PostFinance\DirectLink;
+
+use PostFinance\AbstractPaymentRequest;
+use PostFinance\ShaComposer\ShaComposer;
+use InvalidArgumentException;
+
+class DirectLinkPaymentRequest extends AbstractPaymentRequest {
+
+    const TEST = "https://secure.PostFinance.com/ncol/test/orderdirect.asp";
+    const PRODUCTION = "https://secure.PostFinance.com/ncol/prod/orderdirect.asp";
```
Comment (wysow) [inline lib/PostFinance/DirectLink/DirectLinkPaymentRequest.php:21]: please use only lower case letters in this 2 urls

File: lib/PostFinance/DirectLink/DirectLinkQueryRequest.php [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+namespace PostFinance\DirectLink;
+
+use PostFinance\AbstractDirectLinkRequest;
+use PostFinance\ShaComposer\ShaComposer;
+
+class DirectLinkQueryRequest extends AbstractDirectLinkRequest {
+
+    const TEST = "https://secure.ogone.com/ncol/test/querydirect.asp";
+    const PRODUCTION = "https://secure.ogone.com/ncol/prod/querydirect.asp";
```
Comment (wysow) [inline lib/PostFinance/DirectLink/DirectLinkQueryRequest.php]: This 2 urls should be updated to Postfinance ones...

File: lib/PostFinance/Ecommerce/Alias.php [hunk 8]
```diff
@@ -0,0 +1,71 @@
+<?php
+namespace PostFinance\Ecommerce;
```
Comment (wysow) [inline lib/PostFinance/Ecommerce/Alias.php:3]: A blank line should be added before this one

File: lib/PostFinance/Ecommerce/Alias.php [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+    {
+        $this->alias = $alias;
+    }
+
+    public function __toString()
+    {
+        return $this->alias;
+    }
+
+}
```
Comment (wysow) [inline lib/PostFinance/Ecommerce/Alias.php]: A blank line should be added after this one

File: tests/PostFinance/Tests/Ecommerce/EcommercePaymentRequestTest.php [hunk 10]
```diff
@@ -1,9 +1,8 @@
 <?php
 
 /*
- * This file is part of the Wysow PostFinance package.
+ * This file is part of the Marlon PostFinance package.
```
Comment (wysow) [inline tests/PostFinance/Tests/Ecommerce/EcommercePaymentRequestTest.php]: Original version is better here I think...

File: tests/PostFinance/Tests/Ecommerce/EcommercePaymentRequestTest.php [hunk 11]
```diff
@@ -1,9 +1,8 @@
...
  *
- * (c) Gaultier Boniface <gboniface@wysow.fr>
```
Comment (wysow) [inline tests/PostFinance/Tests/Ecommerce/EcommercePaymentRequestTest.php:6]: Please re-add this line...

File: tests/PostFinance/Tests/Ecommerce/EcommercePaymentResponseTest.php [hunk 12]
```diff
@@ -1,9 +1,8 @@
 <?php
 
 /*
- * This file is part of the Wysow PostFinance package.
+ * This file is part of the Marlon PostFinance package.
```
Comment (wysow) [inline tests/PostFinance/Tests/Ecommerce/EcommercePaymentResponseTest.php]: Original version is better here I think...

File: tests/PostFinance/Tests/Ecommerce/EcommercePaymentResponseTest.php [hunk 13]
```diff
@@ -1,9 +1,8 @@
...
  *
- * (c) Gaultier Boniface <gboniface@wysow.fr>
```
Comment (wysow) [inline tests/PostFinance/Tests/Ecommerce/EcommercePaymentResponseTest.php:6]: Please re-add this line

File: tests/PostFinance/Tests/FormGenerator/SimpleFormGeneratorTest.php [hunk 14]
```diff
@@ -1,9 +1,8 @@
 <?php
 
 /*
- * This file is part of the Wysow PostFinance package.
+ * This file is part of the Marlon PostFinance package.
```
Comment (wysow) [inline tests/PostFinance/Tests/FormGenerator/SimpleFormGeneratorTest.php]: Same as before...

File: tests/PostFinance/Tests/FormGenerator/SimpleFormGeneratorTest.php [hunk 15]
```diff
@@ -1,9 +1,8 @@
...
  *
- * (c) Gaultier Boniface <gboniface@wysow.fr>
```
Comment (wysow) [inline tests/PostFinance/Tests/FormGenerator/SimpleFormGeneratorTest.php:6]: Same...

File: tests/PostFinance/Tests/OgoneTest.php [hunk 16]
```diff
@@ -1,8 +1,7 @@
 <?php
 /*
- * This file is part of the Wysow PostFinance package.
+ * This file is part of the Marlon PostFinance package.
```
Comment (wysow) [inline tests/PostFinance/Tests/OgoneTest.php]: Same...

File: tests/PostFinance/Tests/OgoneTest.php [hunk 17]
```diff
@@ -1,8 +1,7 @@
...
  *
- * (c) Gaultier Boniface <gboniface@wysow.fr>
```
Comment (wysow) [inline tests/PostFinance/Tests/OgoneTest.php:5]: Same...

File: tests/PostFinance/Tests/ParameterFilter/ShaInParameterFilterTest.php [hunk 18]
```diff
@@ -1,9 +1,8 @@
 <?php
 
 /*
- * This file is part of the Wysow PostFinance package.
+ * This file is part of the Marlon PostFinance package.
```
Comment (wysow) [inline tests/PostFinance/Tests/ParameterFilter/ShaInParameterFilterTest.php]: Same...

File: tests/PostFinance/Tests/ParameterFilter/ShaInParameterFilterTest.php [hunk 19]
```diff
@@ -1,9 +1,8 @@
...
  *
- * (c) Gaultier Boniface <gboniface@wysow.fr>
```
Comment (wysow) [inline tests/PostFinance/Tests/ParameterFilter/ShaInParameterFilterTest.php:6]: Same...

File: tests/PostFinance/Tests/ShaComposer/AllParametersShaComposerTest.php [hunk 20]
```diff
@@ -1,9 +1,8 @@
 <?php
 
 /*
- * This file is part of the Wysow PostFinance package.
+ * This file is part of the Marlon PostFinance package.
```
Comment (wysow) [inline tests/PostFinance/Tests/ShaComposer/AllParametersShaComposerTest.php]: Same...

File: tests/PostFinance/Tests/ShaComposer/AllParametersShaComposerTest.php [hunk 21]
```diff
@@ -1,9 +1,8 @@
...
  *
- * (c) Gaultier Boniface <gboniface@wysow.fr>
```
Comment (wysow) [inline tests/PostFinance/Tests/ShaComposer/AllParametersShaComposerTest.php:6]: Same...

File: tests/PostFinance/Tests/ShaComposer/FakeShaComposer.php [hunk 22]
```diff
@@ -1,9 +1,8 @@
 <?php
 
 /*
- * This file is part of the Wysow PostFinance package.
+ * This file is part of the Marlon PostFinance package.
```
Comment (wysow) [inline tests/PostFinance/Tests/ShaComposer/FakeShaComposer.php]: Same...

File: tests/PostFinance/Tests/ShaComposer/FakeShaComposer.php [hunk 23]
```diff
@@ -1,9 +1,8 @@
...
  *
- * (c) Gaultier Boniface <gboniface@wysow.fr>
```
Comment (wysow) [inline tests/PostFinance/Tests/ShaComposer/FakeShaComposer.php:6]: Same...

File: tests/PostFinance/Tests/ShaComposer/LegacyShaComposerTest.php [hunk 24]
```diff
@@ -1,9 +1,8 @@
 <?php
 
 /*
- * This file is part of the Wysow PostFinance package.
+ * This file is part of the Marlon PostFinance package.
```
Comment (wysow) [inline tests/PostFinance/Tests/ShaComposer/LegacyShaComposerTest.php]: Same...

File: tests/PostFinance/Tests/ShaComposer/LegacyShaComposerTest.php [hunk 25]
```diff
@@ -1,9 +1,8 @@
...
  *
- * (c) Gaultier Boniface <gboniface@wysow.fr>
```
Comment (wysow) [inline tests/PostFinance/Tests/ShaComposer/LegacyShaComposerTest.php:6]: Same...

File: tests/TestCase.php [hunk 26]
```diff
@@ -1,9 +1,8 @@
 <?php
 
 /*
- * This file is part of the Wysow PostFinance package.
+ * This file is part of the Marlon PostFinance package.
```
Comment (wysow) [inline tests/TestCase.php]: Same...

File: tests/TestCase.php [hunk 27]
```diff
@@ -1,9 +1,8 @@
...
  *
- * (c) Gaultier Boniface <gboniface@wysow.fr>
```
Comment (wysow) [inline tests/TestCase.php:6]: Same...

File: tests/bootstrap.php [hunk 28]
```diff
@@ -1,9 +1,8 @@
 <?php
 
 /*
- * This file is part of the Wysow PostFinance package.
+ * This file is part of the Marlon PostFinance package.
```
Comment (wysow) [inline tests/bootstrap.php]: Same...

File: tests/bootstrap.php [hunk 29]
```diff
@@ -1,9 +1,8 @@
...
  *
- * (c) Gaultier Boniface <gboniface@wysow.fr>
```
Comment (wysow) [inline tests/bootstrap.php:6]: Same...

**Comment by wysow** [comment]
@b-alidra Please delete the https://packagist.org/packages/balidra/postfinance library on packagist to avoid duplicate, we will release version 1 of this lib right after this PR will be cleaned and merged and tests added.

Thanks again for your contribution.

**Comment by b-alidra** [comment]
HI @wysow,
I've squashed the update to Marlon commits and removed the packagist lib...

**Pull request merged**